security/medusa/l2/acctype_path_access.c

```diff
@@ -144,62 +150,67 @@ char *medusa_get_path(const struct path *path, const struct qstr *last, int last
 			diff = strlen(slash);
 			*slash = '\0';
 			while (slash >= pathbuf) {
-				*(slash + diff) = *slash;
+				slash[diff] = *slash;
 				slash--;
 			}
 			pathbuf += diff;
 		}
 	}
 
 	return pathbuf;
-}
+ }
 
 /**
- * medusa_path_access - make decision about path redirection
- * @action: path access type string descripion ('mkdir', 'unlink', 'symlink', ...)
- * @path_to_redirect: in/out param; stores pointer to buffer, where is path (string),
- *	which is object of decision. If shouldn't be realised redirection,
- *	@path_to_redirect must be set to NULL using medusa_put_path() function call.
- *	This call will free buffer in correct way and set its argument to NULL.
- *	If @path_to_redirect on return is not NULL, underlaying path string is used
- *	for redirection to it.
+ * medusa_path_access() - make decision about path redirection
+ * @action: path access type string descripion ('mkdir', 'unlink', 'symlink',
+ *          ...)
+ * @path_to_redirect: in/out param; stores pointer to a buffer, where a path is
+ *      stored (string), which is the object of decision. If redirection
+ *      shouldn't be relized, @path_to_redirect must be set to NULL using
+ *      medusa_put_path() function call. This call will free buffer in correct
+ *      way and set its argument to NULL. If @path_to_redirect on return is not
+ *      NULL, underlaying path string is used for redirection to it.
  *
- * Returns decision answer, which can be one of:
- *	MED_FAKE_ALLOW - after return force terminating callers function with SUCCESS
- *	MED_DENY - after return deny executing callers function
- *	other values(s), i.e. MED_ALLOW - after return allow executing callers function
+ * Return: Decision answer, which can be one of:
+ *         %MED_FAKE_ALLOW - force caller function to terminate with a SUCCESS
+ *             after return
+ *	   %MED_DENY - deny executing caller function after return other
+ *	       values(s), i.e. %MED_ALLOW - allow executing caller function
+ *	       after return
```

Comment:
there are three types of returned value, not two!
1) MED_FAKE_ALLOW
2) MED_DENY
3) other value(s), i.e. MED_ALLOW